### DISCUSSION TITLE
Where does the copy command save the Exported file in Wasm

### DISCUSSION THREAD

**Discussion by Venkatasamy** [duckdb/duckdb-wasm#1714] [Q&A]
Hi 

I am using DuckDB-WASM in Angular to query a Parquet to convert its format to JSON to bind it to grid.

I am using the below COPY command to export data from a Parquet file as JSON.

await conn?.send(`COPY (SELECT * FROM sample.parquet) TO 'output.json'`)

when i run this command, I am getting the error "Buffering Missing file: output.json".

I am not able to identify in which memory or directory is this file saved? Can anyone provide details on in which memory or directoy does wasm usually export the files.

**Comment by dude0001** [comment]
By default everything in WASM is in memory. There are exceptions to that if you are using extensions.

When you write to a file like this, it is saved first as a registered (in-memory, virtual) file. Your can write the contents of the file to a file buffer and then use Angular/JavaScript to do what you like with the buffer. See https://shell.duckdb.org/docs/classes/index.AsyncDuckDB.html#registerFileBuffer. There is a popular library for React that has an example of how to do this in React https://github.com/holdenmatt/duckdb-wasm-kit/blob/main/src/files/exportFile.ts. or you can see how they do it in https://shell.duckdb.org at https://github.com/duckdb/duckdb-wasm/blob/58fcb9a46b73eac1abb9b0dee9d7c46d1a84f628/packages/duckdb-wasm-shell/src/shell.ts#L42

The error you are seeing could be from various issues. What version of duckdb-wasm? What bundle and version of the bundle? How are you implementing DuckDB-WASM in Angular and can you show your code? What is the contents of the parquet file? Have you tried a simple example first like `(COPY (SELECT 42 as hello_world) TO 'output.json')`? Or are you able to `summarize sample.parquet` and share  the results?

**Comment by carlopi** [comment]
`Buffering missing file` is a warning, not an error, and in this case it's fine, but given it causes a lot of confusion probably it should just become a console.log() or just be removed and logged only on request.

As @dude0001 says, issue a `.files download output.json` after the command.
